cores/cv32e40p/user_manual/source/apu.rst

```diff
@@ -48,6 +51,32 @@ Connection with the FPU
 
 The CV32E40P sends FP operands over the apu\_operands\_o bus; the decoded RV32F operation as ADD, SUB, MUL, etc through the apu\_op\_o bus; the cast, destination and source formats as well as rounding mode through the apu\_flags\_o bus. The respose is the FPU result and relative output flags as Overflow, Underflow, etc.
 
+The operands are encoded as following:
+
+- apu\_operands\_o\[0\] contains the rs1 operand of a OP-FP instructions
+
+- apu\_operands\_o\[1\] contains the rs2 operand of a OP-FP instructions
+
+- apu\_operands\_o\[2\] contains the rs3 operand of a OP-FP instructions
+
+The flags are encoded as following:
+
+- apu\_flags\_o[14:11]  0
+
+- apu\_flags\_o[10:9]   format for casts (int32)
+
+- apu\_flags\_o[8:6]    format source for float2float conversion or move
+
+- apu\_flags\_o[5:3]    format destination for float2float conversion or move
+
+- apu\_flags\_o[2:0]    rounding mode
+
+
+As the core implements only RV32F, the apu\_flags\_o is set to FP32 and INT32 only mode in the format fields.
```

Comment:
Can you add a remark that this interface is only used when FPU = 1 and specify that the outputs can be ignored and the inputs shall all be tied to 0 otherwise?

Question that is maybe not yet relevant for CV32E40P: If on a future chip other functional units (i.e. other than the FPU) are present on the APU interface, how is the correct unit selected (which signal is used for that)?

Reply:
For the apu_*_i signals: Can you specify whether they are allowed to combinatorially depend on the apu_*_o signals or not? (I ask this because you mention that this interface is like an OBI interface where such dependencies are forbidden)

For ``apu\_gnt\_i`, is a default grant allowed (as on OBI)?